src/api/providers/openai.ts

```diff
@@ -17,7 +17,8 @@
 	constructor(options: ApiHandlerOptions) {
 		this.options = options
 		// Azure API shape slightly differs from the core API shape: https://github.com/openai/openai-node?tab=readme-ov-file#microsoft-azure-openai
-		if (this.options.openAiBaseUrl?.toLowerCase().includes("azure.com")) {
+		const urlHost = new URL(this.options.openAiBaseUrl).host;
+		if (urlHost === "azure.com" || urlHost.endsWith(".azure.com")) {
```

Comment:
https://github.com/RooVetGit/Roo-Cline/security/code-scanning/1

> **Copilot Autofix AI generated a fix**
> 
> To fix the problem, we need to parse the URL and check the host explicitly instead of using a substring check. This ensures that the host is correctly validated and prevents bypassing the check by embedding "azure.com" in different parts of the URL.
> 
> * Parse the URL using the URL constructor to extract the host.
> * Check if the host matches "azure.com" or any of its subdomains.
> * Update the condition on line 20 to use the parsed host for validation.

Reply:
Since Copilot Autofix recommends "azure.com" — probably because a person could spoof that in a cybersecurity context, regardless if it's practical or valid — Perhaps it's better safe than sorry in this circumstance?

@ellipsis-dev — I'm going with Copilot Autofix at https://github.com/RooVetGit/Roo-Cline/security/code-scanning/1 on this one.